### ISSUE TITLE
[BUG/ISSUE] Matlab Fun() issues

### ISSUE THREAD

**Issue by jimmielin** [KineticPreProcessor/KPP#56]
I'm trying to get a small mechanism running with Matlab/Octave but I'm running into a weird bug in the latest `dev` commit.

I have a new Matlab mechanism using `small_strato`, specified by `m_small_strato.kpp`:
```
#MODEL      small_strato
#LANGUAGE   matlab
#DOUBLE     ON
#INTEGRATOR rosenbrock
#DRIVER     general
#JACOBIAN   SPARSE_LU_ROW

{This is the small_strato example from Chapter 2 of the KPP manual}
```

Running `kpp` this generates the code fine but it crashes on `Fun()` - looking at `m_small_strato_Fun.m`, the code is obviously wrong:
```matlab
function  [ Aout ] =  m_small_strato_Fun ( V , F , RCT , Vdot )


% Local variables
% A - Rate for each equation
  
A=zeros(1,length(RCT));
Vdot=zeros(1,length(V));

% Computation of equation rates
   A(1) = RCT(1)*F(2) ;
   A(2) = RCT(2)*V(2)*F(2) ;
   A(3) = RCT(3)*V(3) ;
   A(4) = RCT(4)*V(2)*V(3) ;
   A(5) = RCT(5)*V(3) ;
   A(6) = RCT(6)*V(1)*F(1) ;
   A(7) = RCT(7)*V(1)*V(3) ;
   A(8) = RCT(8)*V(3)*V(4) ;
   A(9) = RCT(9)*V(2)*V(5) ;
   A(10) = RCT(10)*V(5) ;


% Aggregate function
   Vdot(1) = A(5)-A(6)-A(7) ;
   Vdot(2) = 2*A(1)-A(2)+A(3)-A(4)+A(6)-A(9)+A(10) ;
   Vdot(3) = A(2)-A(3)-A(4)-A(5)-A(7)-A(8) ;
   Vdot(4) = -A(8)+A(9)+A(10) ;
   Vdot(5) = A(8)-A(9)-A(10) ;

return

% End of Fun function
```

For some reason, `Aout` is set as the output argument. This is wrong. Looking at PACT-1D-HALOGENS' current code at https://github.com/PACT1D/PACT-1D-HALOGENS/blob/main/mechanism/mech_Fun.m

the declaration of `Fun()` should only have `(V, F, RCT)` and `Vdot` is the output argument:
```matlab
function  [ Vdot ] =  mech_Fun ( V , F , RCT )


% Local variables                                                  
% A - Rate for each equation                                       
  
 
A=zeros(1,length(RCT)); 
Vdot=zeros(1,length(V)); 
...
```

I suspect that the function declaration at this line is not correct for MATLAB, but I'm not sure how to fix this. For some reason

```c
  if( z_useAggregate ) {
    FunctionBegin( F_VAR, V, F, RCT, Vdot, Aout );
  }
  else {
    FunctionBegin( FSPLIT_VAR, V, F, RCT, Vdot, P_VAR, D_VAR, Aout );
  }
```

Results in matlab generated code to have `Aout` as the output arg, not `Vdot`. I think `Vdot` is expected since `Fun_Chem()` uses it:

```matlab
%  This line calls the Matlab ODE function routine  
  P = m_small_strato_Fun( Y, FIX, RCONST );
```

**Comment by jimmielin** [comment]
For some reason, `Fun_Chem()` gives a row vector which throws an error in the ODE solver:

```
>> run("m_small_strato_Main.m")
Initial Mass = 1.6240e+17

P =

   1.0e+06 *

   -2.0294    6.2742    2.0851    1.6565   -1.6565

Error using odearguments (line 93)
M_SMALL_STRATO_FUN_CHEM must return a column vector.
```

This seemed to be fine before, maybe it's a recent Matlab update? `Vdot`'s declaration has been `Vdot=zeros(1,length(V));` for quite some time, so it looks more like a MATLAB change.

**Comment by RolfSander** [comment]
Sorry, @jimmielin, I hardly know anything about Matlab. Maybe we should ask our Matlab colleagues directly here?

**Comment by jimmielin** [comment]
Hi Rolf, thanks for the suggestion, I'm still looking through the changes we made for the Matlab fixes for PACT-1D as they seem to be related. I'll follow up in that thread shortly once I figure out what changes are needed and confirm they don't break PACT-1D.

**Comment by yantosca** [comment]
We can now close this issue since PR #99 has been merged into version 3.1.1.
